### PR TITLE
Add an `Event Log Activity` Class

### PR DESCRIPTION
**Pull request by mikeradka**
#### Related Issue: 
#839 Event Log Activity eg 'Audit Log Cleared'

#### Description of changes:

This PR introduces a new class designed for actions related to monitoring the System Event Log Service.

image

### CODE REVIEW

**Comment by jasonbreimer** [comment]
What about a "set-log" activity id? I'm looking at the various detections written against event log activities and I see the concept of event log configuration setting modification. An entity can modify settings regarding the event log using wevtutil command set-log.

**Comment by jasonbreimer** [comment]
Another thought...when looking at clear event log it looks like there are return code status values. This list is oddly limited linked below but I see "success" and "privilege missing". Do we want to capture the successful event or get down to this next layer of capturing a failure by adding some kind of status/return code for each activity id?

https://learn.microsoft.com/en-us/previous-versions/windows/desktop/eventlogprov/cleareventlog-method-in-class-win32-nteventlogfile

**Comment by mikeradka** [comment]
> Another thought...when looking at clear event log it looks like there are return code status values. This list is oddly limited linked below but I see "success" and "privilege missing". Do we want to capture the successful event or get down to this next layer of capturing a failure by adding some kind of status/return code for each activity id?
> 
> https://learn.microsoft.com/en-us/previous-versions/windows/desktop/eventlogprov/cleareventlog-method-in-class-win32-nteventlogfile

That is a good find! Thanks for the suggestion - will take a look at what is possible, it seems like this may be good context to capture.

**Comment by mikeradka** [comment]
For some reason this is failing for `unrecognized keys`:

```TESTING: There are no unrecognized keys
   ERROR: Unrecognized key `actor` of `OcsfAttr` at `attributes.activity_id` in events/system/event_log.json
   ERROR: Unrecognized key `log_file` of `OcsfAttr` at `attributes.activity_id` in events/system/event_log.json
   ERROR: Unrecognized key `log_name` of `OcsfAttr` at `attributes.activity_id` in events/system/event_log.json
   ERROR: Unrecognized key `log_provider` of `OcsfAttr` at `attributes.activity_id` in events/system/event_log.json```

**Comment by mikeradka** [comment]
Created Issue #1047 to address the discrepancy in validation failure output.

**Comment by mikeradka** [comment]
> What about a "set-log" activity id? I'm looking at the various detections written against event log activities and I see the concept of event log configuration setting modification. An entity can modify settings regarding the event log using wevtutil command set-log.

This is a good suggestion, @jasonbreimer. I took a look at the [Microsoft wevtutil documentation](https://learn.microsoft.com/en-us/windows-server/administration/windows-commands/wevtutil), and it notes:

> Enables you to retrieve information about event logs and publishers. You can also use this command to install and **uninstall event manifests**, to **run queries**, and to **export**, **archive**, and **clear** logs.

The `Archive`, and `Clear` actions are covered, and I've added `Export` to the actions. However, I am not yet certain on how we should approach `uninstall event manifests` and `run queries`. Maybe `Uninstall` and `Query` actions are sufficient, but I would like to get some further inputs.

**Comment by pagbabian-splunk** [comment]
> > What about a "set-log" activity id? I'm looking at the various detections written against event log activities and I see the concept of event log configuration setting modification. An entity can modify settings regarding the event log using wevtutil command set-log.
> 
> This is a good suggestion, @jasonbreimer. I took a look at the [Microsoft wevtutil documentation](https://learn.microsoft.com/en-us/windows-server/administration/windows-commands/wevtutil), and it notes:
> 
> > Enables you to retrieve information about event logs and publishers. You can also use this command to install and **uninstall event manifests**, to **run queries**, and to **export**, **archive**, and **clear** logs.
> 
> The `Archive`, and `Clear` actions are covered, and I've added `Export` to the actions. However, I am not yet certain on how we should approach `uninstall event manifests` and `run queries`. Maybe `Uninstall` and `Query` actions are sufficient, but I would like to get some further inputs.
> 
> @floydtree, @pagbabian-splunk, @k2niner any thoughts on whether `Uninstall` and `Query` could work as activities for this new class?

If we don't yet know how to address these two activities, we can hold off until we do, and add them later.  Unless @jasonbreimer or others have a use case of course.  I haven't researched it so I don't have any informed suggestion at the moment.

File: CHANGELOG.md [hunk 1]
```diff
@@ -34,6 +34,7 @@ Thankyou! -->
     16. Added `Session Query` event class. #967
     17. Added `User Query` event class. #967
     18. Added `Tunnel Activity` event class. #1012
+    19. Added `Event Log Activity` event class. #1014
```
Comment (floydtree) [inline CHANGELOG.md:37]: This will need to be moved to the `Unreleased` section

**Review by floydtree — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (mikeradka) [reply CHANGELOG.md]: Nice catch @floydtree . I've updated the `CHANGELOG`, please take a look

File: events/system/event_log.json [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+      "group": "primary",
+      "requirement": "recommended"
+    },
+    "log_name": {
+      "description": "The name of the event log <p style='display:inline;color:red'>targeted by</p> the activity. Example: Windows <code>Security</code>.",
+      "group": "primary",
+      "requirement": "recommended"
+    },
+    "log_provider": {
+      "description": "The logging provider or logging service <p style='display:inline;color:red'>targeted by</p> the activity.<br />Example: <code>Microsoft-Windows-Security-Auditing</code>, <code>Auditd</code>, or <code>Syslog</code>.",
```
Comment (floydtree) [inline events/system/event_log.json:80]: Nice idea to highlight specific strings in the desc. Something to incorporate throughout the schema.

**Review by floydtree — APPROVED** [review]

File: events/system/event_log.json (hunk 2 above)
Comment (mikeradka) [reply events/system/event_log.json:80]: That was all @zschmerber 's suggestion!

**Review by rmouritzen-splunk — APPROVED** [review]

**Pull request merged**